src/Compilers/Core/Portable/DiagnosticAnalyzer/AsyncQueue.cs

```diff
@@ -88,7 +88,10 @@ private bool EnqueueCore(TElement value)
                 waiter = _waiters.Dequeue();
             }
 
-            waiter.SetResult(value);
+            // Invoke SetResult on a separate task, as this invocation could cause the underlying task to executing,
+            // which could be a long running operation that can potentially cause a deadlock if executed on the current thread.
+            Task.Run(() => waiter.SetResult(value));
```

Comment:
Thanks Stephen. I have merged the PR as is in master branch as this was blocking our Jenkins runs. I will do the cleanup as you recommended in a separate PR.